packages/@secretlint/secretlint-rule-sendgrid/src/index.ts

```diff
@@ -0,0 +1,80 @@
+import {
+    SecretLintRuleContext,
+    SecretLintRuleCreator,
+    SecretLintRuleMessageTranslate,
+    SecretLintSourceCode,
+} from "@secretlint/types";
+import { matchPatterns } from "@textlint/regexp-string-matcher";
+
+require("string.prototype.matchall").shim();
+
+export const messages = {
+    SENDGRID_KEY: {
+        en: (props: { KEY: string }) => `found Sendgrid api key: ${props.KEY}`,
+        ja: (props: { KEY: string }) => `Sendgrid APIキーが見つかりました： ${props.KEY}`, // TODO: Needs review.
+    },
+};
+
+export type Options = {
+    /**
+     * Define allow pattern written by RegReg-like strings
+     * See https://github.com/textlint/regexp-string-matcher#regexp-like-string
+     **/
+    allows?: string[];
+};
+
+function reportIfFoundKey({
+    source,
+    options,
+    context,
+    t,
+}: {
+    source: SecretLintSourceCode;
+    options: Required<Options>;
+    context: SecretLintRuleContext;
+    t: SecretLintRuleMessageTranslate<typeof messages>;
+}) {
+    const SENDGRID_KEY_PATTERN = /SG.\w+.\w+-\w+/g;
```

Comment:
```suggestion
    const SENDGRID_KEY_PATTERN = /SG\.\w+\.\w+-\w+/g;
```

Maybe `.` should be escaped, becase `.` match any charater.

Also, Can we limit the `\w` lengths for clarity?

```js
const SENDGRID_KEY_PATTERN = /SG\.\w{1,128}\.\w{1,128}-\w{1,128}/g
```